### PR TITLE
docs: Document dependency upgrade procedure

### PR DESCRIPTION
**Pull request by nevzheng**
Document Dependency upgrade procedure.

Fixes: https://github.com/hashicorp/terraform-provider-google/issues/15583

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/run-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none
```

### CODE REVIEW

File: docs/content/reference/upgrade_dependencies.html.markdown [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Upgrading Dependencies
+
+## Procedure
+
+1. Please have an instance of `magic-modules` available. It will be used later in this procedure.
```
Comment (melinath) [inline docs/content/reference/upgrade_dependencies.html.markdown:5]: [Avoid please](https://developers.google.com/style/tone#politeness-and-use-of-please) in documentation. Additionally, setup like this should be in a [Before you begin](https://cloud.google.com/guides/docs/task#document_the_prerequisites) section.

```suggestion
## Before you begin

1. [Set up your development environment]({{< ref "/get-started/generate-providers/" >}})
```

File: docs/content/reference/upgrade_dependencies.html.markdown [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Upgrading Dependencies
+
+## Procedure
```
Comment (melinath) [inline docs/content/reference/upgrade_dependencies.html.markdown:3]:
```suggestion
```

File: docs/content/reference/upgrade_dependencies.html.markdown [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Upgrading Dependencies
```
Comment (melinath) [inline docs/content/reference/upgrade_dependencies.html.markdown:1]: There needs to be [front matter](https://gohugo.io/content-management/front-matter/) here. Additionally, see [Headings and titles](https://developers.google.com/style/headings) for guidance on writing headings.

Weight 65 will put this just above "Promote to GA"

```suggestion
---
title: Upgrade provider dependencies
weight: 65
---
# Upgrade provider dependencies
```

File: docs/content/reference/upgrade_dependencies.html.markdown [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+## Procedure
+
+1. Please have an instance of `magic-modules` available. It will be used later in this procedure.
+2. Checkout & `cd` to [`terraform-provider-google-beta`](http://github.com/hashicorp/terraform-provider-google-beta)
+3. Upgrade your desired package. EX: `go get -u google.golang.org/api` You may need to upgrade all packages at once with `go get -u ./..`. [More Info](https://golang.cafe/blog/how-to-upgrade-golang-dependencies.html).
```
Comment (melinath) [inline docs/content/reference/upgrade_dependencies.html.markdown:7]: See https://developers.google.com/style/format-examples for style guidance on examples, and https://developers.google.com/style/procedures for information about how to format procedure lists that include code.

File: docs/content/reference/upgrade_dependencies.html.markdown [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+4. Run `make lint`. If there is an error saying `go get` should be run to fill in missing dependencies, please do so. Repeat until `make lint` passes.
+5. Copy `go.mod` to `magic-modules` at `mmv1/third_party/terraform/go.mod.erb`. Do not include the `module github.com/hashicorp/terraform-provider-google-beta` beginning line that should stay as `module github.com/hashicorp/terraform-provider-google`.
+6. Copy `go.sum` to `magic-modules` at `mmv1/third_party/terraform/go.sum`.
+7. `git clean -f -d && git reset --hard` in beta provider
+8. `git clean -f -d && git reset --hard` in ga provider
+9. Generate the GA Provider. `make lint` should pass.
+10. Generate the Beta Provider. `make lint` should pass.
+11. Next Unit and Acceptance test should pass. The easiest way to do so is to
```
Comment (melinath) [inline docs/content/reference/upgrade_dependencies.html.markdown:15]: See https://cloud.google.com/guides/components/whats-next for more information about "What's next" links.

```suggestion
## What's next?

- [Run tests]({{< ref "/develop/run-tests" >}}
- [Create a pull request](https://docs.github.com/en/get-started/quickstart/github-flow#create-a-pull-request)
```

**Review by melinath — CHANGES_REQUESTED** [review]
Thanks for your contribution! This does not currently match our style guide; it should look more similar to https://googlecloudplatform.github.io/magic-modules/get-started/generate-providers/ Please read through the following style guides and rewrite this guide accordingly:

- https://cloud.google.com/guides/docs/task
- https://developers.google.com/style/

File: docs/content/reference/upgrade_dependencies.html.markdown (hunk 3 above)
Comment (nevzheng) [reply docs/content/reference/upgrade_dependencies.html.markdown:1]: done.

File: docs/content/reference/upgrade_dependencies.html.markdown (hunk 5 above)
Comment (nevzheng) [reply docs/content/reference/upgrade_dependencies.html.markdown:15]: done.

File: docs/content/reference/upgrade_dependencies.html.markdown (hunk 1 above)
Comment (nevzheng) [reply docs/content/reference/upgrade_dependencies.html.markdown]: done

File: docs/content/reference/upgrade_dependencies.html.markdown (hunk 2 above)
Comment (nevzheng) [reply docs/content/reference/upgrade_dependencies.html.markdown]: done

File: docs/content/reference/upgrade_dependencies.html.markdown (hunk 4 above)
Comment (nevzheng) [reply docs/content/reference/upgrade_dependencies.html.markdown]: i hope i improved it.

File: docs/content/reference/upgrade_dependencies.html.markdown (hunk 5 above)
Comment (nevzheng) [reply docs/content/reference/upgrade_dependencies.html.markdown]: done.

**Pull request closed without merging**